learning/01_Exercises/Lab 7 to 9/services/controller.py

```diff
@@ -0,0 +1,161 @@
+            self.clientsRepository.save(Client)
+            print(f"\n{Client} has been added to the list.")
+        else:
+            print(f"The {Client}'s details appears to be wrong, please try again.")
+
+    # REMOVE SECTION
+    def removeMovie(self, ID):
+        if ID in self.moviesRepository.getIdList():
+            newMovieList = self.moviesRepository.remove(ID)
+            self.moviesRepository.removeAll(), self.moviesRepository.saveAll(newMovieList)
+            print(f"\nThe movie with {ID} id has been removed.")
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    def removeClient(self, ID):
+        if ID in self.clientsRepository.getIdList():
+            newClientList = self.clientsRepository.remove(ID)
+            self.clientsRepository.removeAll(), self.clientsRepository.saveAll(newClientList)
+            print(f"\nThe client with {ID} id has been removed.")
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    # UPDATE SECTION
+    def updateMovie(self, ID, newMovie):
+        if self.validatorForMovie.checkID(ID, self.moviesRepository.getIdList()):
+            self.moviesRepository.update(ID, newMovie)
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    def updateClient(self, ID, newClient):
+        if self.validatorForMovie.checkID(ID, self.clientsRepository.getIdList()):
+            self.clientsRepository.update(ID, newClient)
+        else:
+            print(f"\nThe ID: {ID} is invalid! Please try again.")
+
+    # SEARCH SECTION
+    def searchMovies(self, findMovie):
+        if findMovie in self.moviesRepository.getNames():
+            for movie in self.moviesRepository.getAll():
+                if movie.title == findMovie:
+                    print(f"{findMovie} | Found! | Full information: ")
+                    print(f"{movie}")
+        else:
+            print(f"{findMovie} | Could not been found! Please try again.")
+
+    def searchClients(self, findClient):
+        if findClient in self.clientsRepository.getNames():
+            for client in self.clientsRepository.getAll():
+                if client.name == findClient:
+                    print(f"{findClient} | Found! | Full information: ")
+                    print(f"{client}")
+        else:
+            print(f"{findClient} | Could not been found! Please try again.")
+
+    # RENT SECTION
+    def rentMovies(self, theClient, theMovie):
+        if theClient in self.clientsRepository.getNames():
+            if theMovie in self.moviesRepository.getNames():
+                if self.validatorForMovie.checkAvailability(self.moviesRepository.getMovie(theMovie).availability):
+                    self.clientsRepository.saveMovie(theClient, self.moviesRepository.getMovie(theMovie).title)
+                    self.moviesRepository.updateStatus(self.moviesRepository.getMovie(theMovie))
```

Comment:
This function should not exist, you should use `update` from the repository and update the movie here, inside the controller

Reply:
The function/method is about renting movies. I think that I could improve those variables names so its clear.
It updates the availability status from Available to Not Available.